bin/mailmap_check.py

```diff
@@ -0,0 +1,297 @@
+    parser.add_argument('--update-authors', action='store_true',
+            help=filldedent("""Also update the AUTHORS file. Note that it
+            should only necessary for the release manager to do this as part of
+            the release process for SymPy."""))
+    args = parser.parse_args(args)
+
+    if not check_git_version():
+        return 1
+
+    # find who git knows ahout
+    try:
+        git_people = get_authors_from_git()
+    except AssertionError as msg:
+        print(red(msg))
+        return 1
+
+    lines_mailmap = read_lines(mailmap_path())
+
+    def key(line):
+        # return lower case first address on line or
+        # raise an error if not an entry
+        if '#' in line:
+            line = line.split('#')[0]
+        L, R = line.count("<"), line.count(">")
+        assert L == R and L in (1, 2)
+        return line.split(">", 1)[0].split("<")[1].lower()
+
+    who = OrderedDict()
+    for i, line in enumerate(lines_mailmap):
+        try:
+            who.setdefault(key(line), []).append(line)
+        except AssertionError:
+            who[i] = [line]
+
+    problems = False
+    missing = False
+    ambiguous = False
+    dups = defaultdict(list)
+
+    for person in git_people:
+        email = key(person)
+        dups[email].append(person)
+        if email not in who:
+            print(red("This author is not included in the .mailmap file:"""))
+            print(person)
+            missing = True
+        elif not any(p.startswith(person) for p in who[email]):
+            print(red("Ambiguous names in .mailmap"))
+            print(red("This email address appears for multiple entries:"))
+            print('Person:', person)
+            print('Mailmap entries:')
+            for line in who[email]:
+                print(line)
+            ambiguous = True
+
+    if missing:
+        print(red("""
+        The .mailmap file needs to be updated because there are commits with
+        unrecognised author/email metadata.
+        """))
+        problems = True
+
+    if ambiguous:
+        print(red("""
+        Lines should be added to .mailmap to indicate the correct name and
+        email aliases for all commits.
+        """))
+        problems = True
+
+    for email, commitauthors in dups.items():
+        if len(commitauthors) > 2:
+            print(red("""
+            The following commits are recorded with different metadata but the
+            same/ambiguous email address. The .mailmap file will need to be
+            updated."""))
+            problems = True
+
+    lines_mailmap_sorted = sort_lines_mailmap(lines_mailmap)
+    write_lines(mailmap_path(), lines_mailmap_sorted)
+
+    if lines_mailmap_sorted != lines_mailmap:
+        problems = True
+        print(red("The mailmap file was reordered"))
+
+    if problems:
+        print(red("""
+        For instructions on updating the .mailmap file see:
+        https://github.com/sympy/sympy/wiki/Development-workflow"""))
```

Comment:
Even if we don't add it now, we can add the anchor for the subheading on this page that will contain the instructions, like https://github.com/sympy/sympy/wiki/Development-workflow#update-mailmap

Reply:
> It's pretty easy to use GitHub Actions to do things like comment on the PR and add a commit to the PR (or master).

No, this can't be automated. The contributor has to say what name they want in the AUTHORS file. They also should be expected top understand the implication that their name is going into the AUTHORS file and why the name/email metadata in their commits is important. This is not something that anyone else can do on behalf of the contributor.

Reply:
I've added that.